src/orquestra/sdk/_ray/_dag.py

```diff
@@ -382,7 +382,9 @@ def _gather_kwargs(
 def _make_ray_dag(
     client: RayClient, wf: ir.WorkflowDef, wf_run_id: str, project_dir: t.Optional[Path]
 ):
-    ray_consts: t.Dict[ir.ConstantNodeId, t.Any] = wf.constant_nodes
+    # We need to modify the dictionary, by adding new key-values.
+    # A shallow copy is OK, as long as we don't modify the _values_ in the dict.
+    ray_consts: t.Dict[ir.ConstantNodeId, t.Any] = wf.constant_nodes.copy()
```

Comment:
Question, non-blocking: would it cost much to make deep-copy just-in-case? Its more error-prone in the future in case we missed whats reference and whats copy. Up to you to decide, not saying it is must-have

Reply:
After discussing on Slack, we came to the conclusion that shallow copy is OK for now.

Deep copies seemed to be expensive when dealing with workflows with pickled constants (in particular).

I updated the comment to address that we should NOT modify values from the dict.